### PR TITLE
Several bug fix on cart related with products stock

### LINKED ISSUE #488: [FO] CART- When your add more items to cart than is in stock there are few issues
### Describe the bug

When your add more items to cart than is in stock there are few issues

### Steps to reproduce

1. BO > Update the `Mug The best is yet to come` quantity to 1 and save
2. FO > Add one `Mug The best is yet to come` to the Cart
3. Access to the cart page
4. Try to add more of `Mug The best is yet to come` to the cart 
5. You can see the message `The available purchase order quantity for this product is 1.` displayed, but you don't know how many pieces are in stock.
6. Change the quantity, the error disappears, only a toast is shown in top right.
7. Even though there is an error the `Proceed to checkout` is active.
8. When you click it, you stay on the same page, but the red error disappears.
9. When you add another product that is in stock, go to cart, delete this instock product, the red error disappears.

**Screen record**
[error-msg-cart.webm](https://github.com/PrestaShop/hummingbird/assets/23308427/d00b89a1-8d50-41f0-bce4-bf10acd52fdc)

### Additional infos

- HummingBird version: 0.1.5
- PS version: develop
- PHP version: 8.0

### PR DESCRIPTION
**Pull request by GytisZum**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | There is several bug fixes. First of them is "Proceed to checkout" button disabling in cart page if order is not available and product has not enough quantity. Second is about the toast color which changed from yellow (warning) to red (danger).  Third. The product quantity in input and in cart is set to max if customer has more then it exist in stock. This fix will not allow customer to add more products into cart if quantity is more then in stock and allow out-of-stock products is disabled is and both input and cart is falling back into previous value. As well this fix contains console error fix.
| Type?             | bug fix
| BC breaks?        |no
| Deprecations?     | no
| Fixed ticket?     | Fixes #488
| Sponsor company   | PrestaShop Corp.
### Button Disabled:
Steps:
1. Go to back office and set "Mug Today is a good day" quantity to 1 and allow out-of-stock product to order.
2. Go to front office and add more then 1 "Mug Today is a good day" 
3. Go back to back office and deny out-of-stock product to order.
4. Go back to front office refresh the page and see the disabled button.
image

### Toast Color:
1. Go to back office and set "Mug Today is a good day" quantity to 1 and deny out-of-stock product to order.
2. Go to front office and add more then 1 "Mug Today is a good day" 
3. Go to cart and increase quantity of that product.
4. The toast color should be red.
image

### Product quantity fix:
1. Go to back office and set "Mug Today is a good day" quantity to 1 and deny out-of-stock product to order.
2. Go to front office and add more then 1 "Mug Today is a good day" 
3. Go to cart and increase quantity of that product.
4. Then amount in cart is exceeded with current stock amount you should see the toast and input value which will change back to previous.

### Bonus fix:
Console error:
image

This error is occurred due to loading element which does not exist yet, but it's trying to initiate it on checkout pages. This error related with carrier extra content on checkout.
To test it need to check if carrier extra content works as previously.

#### Environment:
Hummingbird version: develop
PrestaShop version 8.1.0

### CODE REVIEW

File: src/js/components/useQuantityInput.ts
Comment (GytisZum) [reply src/js/components/useQuantityInput.ts]: Thank you for the suggestion, your solution looks more readable :fire:

**Review by M0rgan01 — APPROVED** [review]

**Review by ga-devfront — APPROVED** [review]
all it's good for me, thank you for your contribution !

**Review by kpodemski — APPROVED** [review]

**Review by sallemiines — COMMENTED** [review]
Hello @GytisZum 
Thanx for this PR ! 

I tried to check this PR , and here is some feedback : 

- Proceed to checkout" button is not disable in the cart page if order is not available and product has not enough quantityif order is not available and product has not enough quantity :x: 

-  Toast color is changed  to red :ok: but the order quantity for this product not exist :x:  
You can check the attached screen recorder 

https://github.com/PrestaShop/hummingbird/assets/31663369/b2639e7e-8dbd-4e76-ab63-44f1d8ee8a7b



- The amount in cart is exceeded with current stock amount you should see the toast and input value which will change back to previous. :x: 

You can check the attached screen recorder 

https://github.com/PrestaShop/hummingbird/assets/31663369/64784187-ddcc-45e7-a36e-2cbe079674bf

Please check & feedback ! 
Thank y !

**Review by sallemiines — COMMENTED** [review]
Hello @GytisZum 
Thanx for this PR ! 

I tried to check this PR , and here is some feedback : 

- Proceed to checkout" button is not disable in the cart page if order is not available and product has not enough quantityif order is not available and product has not enough quantity :x: 

-  Toast color is changed  to red :ok: but the order quantity for this product not exist :x:  
You can check the attached screen recorder 

https://github.com/PrestaShop/hummingbird/assets/31663369/b2639e7e-8dbd-4e76-ab63-44f1d8ee8a7b



- The amount in cart is exceeded with current stock amount you should see the toast and input value which will change back to previous. :x: 

You can check the attached screen recorder 

https://github.com/PrestaShop/hummingbird/assets/31663369/64784187-ddcc-45e7-a36e-2cbe079674bf

Please check & feedback ! 
Thank y !

**Comment by GytisZum** [comment]
> 

Hello,

thanks you for testing the fixes. However I see that you are using PrestaShop version 1.7.8.9. 
We already know that older PrestaShop version works differently and we were informed that we need to be compatible only from PrestaShop 8.1.0

**Comment by sallemiines** [comment]
> 1. Mug Today is a good day


Thnx for your feedback ! 


I re-test the PR on PS 810 as you mentioned before

- **Proceed to checkout**" button is not disable in the cart page if order is not available and product has not enough quantityif order is not available and product has not enough quantity :heavy_check_mark: 

- Toast color is changed to red and  the order quantity for this product  :heavy_check_mark:  but it appears and disappear quickly in the wrong placement :x: 

You can check the attached screen recorder


https://github.com/PrestaShop/hummingbird/assets/31663369/d76583af-b774-4207-9552-43d0e2d3f457

Waiting for yr feedback ! 

Thank you

**Comment by GytisZum** [comment]
> > 1. Mug Today is a good day
> 
> Thnx for your feedback !
> 
> I re-test the PR on PS 810 as you mentioned before
> 
>     * **Proceed to checkout**" button is not disable in the cart page if order is not available and product has not enough quantityif order is not available and product has not enough quantity heavy_check_mark
> 
>     * Toast color is changed to red and  the order quantity for this product  heavy_check_mark  but it appears and disappear quickly in the wrong placement x
> 
> 
> You can check the attached screen recorder
> Mug.Today.is.a.good.day.mp4
> 
> Waiting for yr feedback !
> 
> Thank you

Hello again,

Thank You for Your Feedback !!
I did several tests with different PrestaShop versions and I was able to recreate your issue with PrestaShop [8.1.0-beta.1](https://github.com/PrestaShop/PrestaShop/releases/tag/8.1.0-beta.1)

After comparison I found that real candidate version has changes in cart page which affects the behavior of the error messages.

I did some communication with the core team and got answer that tests must be done on stable version which does not exist at the moment, but closest one is PrestaShop version [8.1.0-rc.1](https://github.com/PrestaShop/PrestaShop/releases/tag/8.1.0-rc.1).

Sorry I was not precise enough, but maybe you can test it on more stable version ?? :) 

Waiting for the feedback 
Thank you

**Comment by kpodemski** [comment]
@sallemiines you can use the same build that QA tests for the final 8.1 release

**Review by aniszr — APPROVED** [review]
Hello @GytisZum 

Thanks for your PR, I tested it with 8.1.0 build 2 & Develop .

- Proceed to checkout" button is not disable in the cart page if order is not available and product has not enough quantityif order is not available and product has not enough quantity :heavy_check_mark: 

- Toast color is changed to red and the order quantity for this product :heavy_check_mark: 
-  Amount in cart is exceeded with current stock amount you should see the toast and input value which will change back to previous. :heavy_check_mark: 
 

LGTM, QA :heavy_check_mark:

See attached screenshots:
image
image

https://github.com/PrestaShop/hummingbird/assets/92626242/c59693e4-62fb-40e2-adcb-561696853e28


Thanks!

**Comment by Progi1984** [comment]
Ping @Hlavtox for merge :wink:

**Comment by kpodemski** [comment]
thanks @GytisZum

**Pull request merged**
